scripts/ci/pr-bot/gatherMetrics.ts

```diff
@@ -0,0 +1,527 @@
+      aggregatedStats.averageTimeToNewContributorFirstReview
+    )}`;
+
+    console.log(
+      "Average time in minutes from PR creation to completion:",
+      convertMsToRoundedMinutes(aggregatedStats.averageTimeCreationToCompletion)
+    );
+    csvOutput += `,${convertMsToRoundedMinutes(
+      aggregatedStats.averageTimeCreationToCompletion
+    )}`;
+
+    console.log(
+      "Total number of reviewers:",
+      aggregatedStats.numUsersPerformingReviews
+    );
+    csvOutput += `,${aggregatedStats.numUsersPerformingReviews}`;
+
+    console.log(
+      "Total number of committers performing reviews:",
+      aggregatedStats.numCommittersPerformingReviews
+    );
+    csvOutput += `,${aggregatedStats.numCommittersPerformingReviews}`;
+
+    console.log(
+      "Total number of non-committers performing reviews:",
+      aggregatedStats.numNonCommittersPerformingReviews
+    );
+    csvOutput += `,${aggregatedStats.numNonCommittersPerformingReviews}`;
+
+    console.log(
+      "Gini index (fairness) of committers performing reviews:",
+      aggregatedStats.giniIndexCommittersPerformingReviews
+    );
+    csvOutput += `,${aggregatedStats.giniIndexCommittersPerformingReviews}`;
+
+    console.log(
+      "Average time in minutes from committer assignment to PR merge:",
+      convertMsToRoundedMinutes(
+        aggregatedStats.averageTimeFromCommitterAssignmentToPrMerge
+      )
+    );
+    csvOutput += `,${convertMsToRoundedMinutes(
+      aggregatedStats.averageTimeFromCommitterAssignmentToPrMerge
+    )}`;
+  }
+  fs.writeFileSync(PATH_TO_METRICS_CSV, csvOutput);
+  console.log(`Output written to ${PATH_TO_METRICS_CSV}`);
+}
+
+async function gatherMetrics() {
+  // We will only aggregate metrics from the last 90 days,
+  // but will look further back to determine if this is a user's first contribution
+  const pullsFromLastYear = await getCompletedPullsFromLastYear();
+  let cutoffDate = new Date();
+  cutoffDate.setDate(cutoffDate.getDate() - 90);
+
+  let pullStats: PrStats[] = [];
+  console.log("Extracting stats from pulls - this may take a while");
+  for (let i = 0; i < pullsFromLastYear.length; i++) {
+    let pull = pullsFromLastYear[i];
+    if (new Date(pull.created_at) > cutoffDate && pull.merged_at) {
+      pullStats.push(await extractPrStats(pull, pullsFromLastYear));
+    }
+    if (i % 10 === 0) {
+      process.stdout.write(".");
+    }
+  }
+
+  console.log("\nDone extracting stats, formatting results");
+
+  let statBuckets: { [key: number]: PrStats[] } = {};
+  let bucketEnd = new Date();
+  bucketEnd.setUTCHours(23, 59, 59, 999);
+  pullStats.forEach((pullStat) => {
+    let bucketStart = getMetricBucketStartDate(pullStat, bucketEnd);
+    if (bucketStart in statBuckets) {
+      statBuckets[bucketStart].push(pullStat);
+    } else {
+      statBuckets[bucketStart] = [pullStat];
+    }
+  });
+  await reportMetrics(statBuckets);
+}
+
+gatherMetrics();
+
+export {};
```

Comment:
I'm not familiar with this self executing export pattern. This pattern could make it difficult to debug when an import causes a side-effect. I think it's better to call `gatherMetrics` from the running class.

Reply:
This is a common pattern for github actions, we chatted about it and decided its ok to keep